pkg/dr/health.go

```diff
@@ -0,0 +1,94 @@
+package dr
+
+import (
+	ctx "context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/rook/kubectl-rook-ceph/pkg/exec"
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	rookv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type secretData struct {
+	Key     string `json:"key"`
+	MonHost string `json:"mon_host"`
+}
+
+func Health(context *k8sutil.Context, operatorNamespace, cephClusterNamespace string, args []string) {
+	fmt.Println("INFO: fetching the cephblockpools with mirroring enabled")
+	blockPoolList, err := context.RookClientset.CephV1().CephBlockPools(cephClusterNamespace).List(ctx.TODO(), v1.ListOptions{})
+	if err != nil {
+		log.Error(err)
+	}
+
+	var mirrorBlockPool rookv1.CephBlockPool
+	for _, blockPool := range blockPoolList.Items {
+		if blockPool.Spec.Mirroring.Enabled && blockPool.Spec.Mirroring.Peers != nil {
+			mirrorBlockPool = blockPool
+			fmt.Printf("found %q cephblockpool with mirroring enabled\n", mirrorBlockPool.Name)
+		}
+	}
+
+	if mirrorBlockPool.Name == "" {
+		fmt.Printf("DR is not confiqured, cephblockpool with mirroring enabled not found.\n")
+	}
+
+	secretDetail := mirrorBlockPool.Spec.Mirroring.Peers.SecretNames[0]
+
+	secretData, err := extractSecretName(context, operatorNamespace, cephClusterNamespace, secretDetail)
+	if err != nil {
+		fmt.Printf("failed to extract secret %s", secretDetail)
+		log.Fatal(err)
+	}
+
+	fmt.Println("running ceph status from peer cluster")
+
+	cephArgs := []string{"-s", "--mon-host", secretData.MonHost, "--id", "rbd-mirror-peer", "--key", secretData.Key}
```

Comment:
client-id (`--id`) should be retrieved from the decoded token as well.

Reply:
thanks, should be good now.